platform/src/common/components/Map/components/MapNodes.js

```diff
@@ -103,6 +103,30 @@ export const getAQIcon = (pollutant, value) => {
   }
 };
 
+export const getAQIMessage = (pollutant, value) => {
+  if (!markerDetails.hasOwnProperty(pollutant)) {
+    throw new Error(`Invalid pollutant: ${pollutant}`);
+  }
+
+  const aqiCategory = getAQICategory(pollutant, value);
```

Comment:
Hi @Codebmk , this is amazing. We could refine this code further.....

We can use a combination of a lookup object and the optional chaining operator (?.). This approach simplifies the code by removing the repetitive if-else structure and makes it easier to add or modify conditions in the future. Additionally, the optional chaining operator is supported in modern JavaScript environments,but it's not available in older versions of JavaScript so we might want to rethink things as we go about its usage --- it is supported in Node.js versions 14 and above, and in most modern browsers. However, it's not available in older JavaScript environments.

Here's how we can refactor the code:

```
const aqiMessages = {
 'GoodAir': 'Enjoy the day with confidence in the clean air around you.',
 'ModerateAir': 'Today is a great day for an outdoor activity.',
 'UnhealthyForSensitiveGroups': 'Reduce the intensity of your outdoor activities.',
 'Unhealthy': 'Avoid activities that make you breathe more rapidly. Today is the perfect day to spend indoors reading.',
 'VeryUnhealthy': 'Reduce the intensity of your outdoor activities. Try to stay indoors until the air quality improves.',
 'Hazardous': 'If you have to spend a lot of time outside, disposable masks like the N95 are helpful.',
};

const aqiCategory = getAQICategory(pollutant, value);

return aqiMessages[aqiCategory?.icon] || '';
```

Reply:
Noted